dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -194,6 +198,26 @@ private static Validator createValidator() {
         return validatorFactory.getValidator();
     }
 
+    /**
+     * Decide whether a gitReference is excluded, given a workflow/service's filters
+     * @param gitRefPath Path.of(gitReference) for glob matching with PathMatcher
+     * @param filters Filters specified for a workflow/service in .dockstore.yml
+     * @return
+     */
+    public static boolean filterGitReference(final Path gitRefPath, final List<String> filters) {
+        return filters.isEmpty() || filters.stream().anyMatch(filter -> {
+            String matcherString;
+            // Use regex if filter string is surrounded by /
+            if (filter.matches("^\\/.*\\/$")) {
+                matcherString = "regex:^refs\\/(heads\\/|tags\\/|)" + filter.substring(1, filter.length() - 1);
```

Comment:
Escaping gets confusing; I wonder if anything might get lost when going from the YAML to here.

Let's say I want to only filter tags that use semantic versioning. I would have a regex something like this in the YAML: `\d+\.\d+\.\d+`. Will that work? At first I was thinking it wouldn't, but now I think it would. If I wrote that in Java, I think I would have to escape the backslashes themselves. But that might be only for the compiler, so maybe this does work. Worth adding a test for in any case.

Reply:
Escaping is only really a thing for Java string literals here... need to use `\\` to be interpreted as a `\`.
YAML reading shouldn't behave like this, but I can check with the library we use to read it.

Reply:
Integration test now uses an example with `\d+\.\d+`